Analysis/src/TypeVar.cpp

```diff
@@ -1072,14 +1072,16 @@ static std::vector<TypeId> parseFormatString(TypeChecker& typechecker, const cha
                 continue;
 
             // we just ignore all characters (including flags/precision) up until first alphabetic character
-            while (i < size && !(data[i] > 0 && isalpha(data[i])))
+            while (i < size && !(data[i] > 0 && (isalpha(data[i]) || data[i] == '*')))
                 i++;
 
             if (i == size)
                 break;
 
             if (data[i] == 'q' || data[i] == 's')
                 result.push_back(typechecker.stringType);
+            else if (data[i] == '*')
+                result.push_back(typechecker.errorRecoveryType(typechecker.anyType));
```

Comment:
```suggestion
                result.push_back(typechecker.anyType);
```

Reply:
You actually don't want `any` here. This will be an implementation detail, but push in either a free type or just use `unknown`. Unification doesn't bind a free type to `unknown`, but will do so for `any`.

Reply:
I went with unknown

Reply:
A test that proves this does the right thing, something like this, would be great!

```lua
local function f(x): string
  local foo = string.format("%*", x)
  return x
end
```

where then we expect `f`'s type to be `(string) -> string`.

Reply:
Will get to that today, was unsure the best way to go about it 🙂

Reply:
Do you mean (unknown) -> string, by the way?

Reply:
`(string)->string` is correct, I think the idea here is:

- If `string.format` call is absent, `x` will infer as `string` because the return value of `f` is type-annotated as `string`, as such `f` will infer as `(string)->string`
- If `string.format` call is present and the argument of that has `unknown` type, type inference won't change the type of `x` so the above still holds
- If `string.format` call is present and the argument of that has `any` type, type inference will assign type `any` to `x` and the type of `f` will be `(any)->string`

You can simply use requireType to check this in a test.

Reply:
Didn't notice we were returning back x, makes sense.